AlphaWallet/Tokens/Types/TokenViewModel.swift

```diff
@@ -0,0 +1,140 @@
+//
+//  TokenViewModel.swift
+//  AlphaWallet
+//
+//  Created by Vladyslav Shepitko on 14.07.2022.
+//
+
+import Foundation
+import BigInt
+
+struct TokenViewModel {
+    let contractAddress: AlphaWallet.Address
+    let symbol: String
+    let decimals: Int
+    let server: RPCServer
+    let type: TokenType
+    let name: String
+    let shouldDisplay: Bool
+    let balance: BalanceViewModel
+    let tokenScriptOverrides: TokenScriptOverrides?
+}
+
+struct TokenScriptOverrides {
+    let title: String
+    let titleInPluralForm: String
+    let shortTitleInPluralForm: String
+    let symbolInPluralForm: String
+    let hasNoBaseAssetDefinition: Bool
+    let server: RPCServerOrAny?
+
+    let xmlHandler: XMLHandler
+}
+
+extension TokenScriptOverrides {
+    init(token: Token, assetDefinitionStore: AssetDefinitionStore, sessions: ServerDictionary<WalletSession>, eventsDataStore: NonActivityEventsDataStore) {
+        let xmlHandler = XMLHandler(token: token, assetDefinitionStore: assetDefinitionStore)
+        self.symbolInPluralForm = token.symbolInPluralForm(withAssetDefinitionStore: assetDefinitionStore)
+        self.title = token.title(withAssetDefinitionStore: assetDefinitionStore)
+        //NOTE: replace if needed
+        switch token.type {
+        case .erc20, .nativeCryptocurrency:
+            self.shortTitleInPluralForm = token.shortTitleInPluralForm(withAssetDefinitionStore: assetDefinitionStore)
+            self.titleInPluralForm = token.titleInPluralForm(withAssetDefinitionStore: assetDefinitionStore)
+        case .erc875, .erc1155, .erc721, .erc721ForTickets:
+            self.shortTitleInPluralForm = token.shortTitleInPluralForm(withAssetDefinitionStore: assetDefinitionStore, eventsDataStore: eventsDataStore, forWallet: sessions.anyValue.account)
+            self.titleInPluralForm = token.titleInPluralForm(withAssetDefinitionStore: assetDefinitionStore, eventsDataStore: eventsDataStore, forWallet: sessions.anyValue.account) ?? token.titleInPluralForm(withAssetDefinitionStore: assetDefinitionStore)
+        }
+
+        self.xmlHandler = xmlHandler
+        self.hasNoBaseAssetDefinition = xmlHandler.hasNoBaseAssetDefinition
+        self.server = xmlHandler.server
+    }
+}
+
+extension TokenScriptOverrides: Hashable {
+    static func == (lhs: TokenScriptOverrides, rhs: TokenScriptOverrides) -> Bool {
+        return lhs.shortTitleInPluralForm == rhs.shortTitleInPluralForm && lhs.symbolInPluralForm == rhs.symbolInPluralForm && lhs.title == rhs.title && lhs.titleInPluralForm == rhs.titleInPluralForm && lhs.hasNoBaseAssetDefinition == rhs.hasNoBaseAssetDefinition && lhs.server == rhs.server
+    }
+
+    func hash(into hasher: inout Hasher) {
+        //FIXME: do we need to add fields here?
```

Comment:
Need to fix this?

Reply:
btw i quite don't know what fields do we need to add here, any suggestions?

Reply:
Is this to support diffs? If so, the `==` is more relevant, not `hash(into:)`. Refer https://github.com/AlphaWallet/alpha-wallet-ios/pull/4933#discussion_r921865706

Reply:
we check for diffs in view models, actually what we display on view, `ethCellViewModel`, .. .etc, but not for for actual applying comparison for objects. I would like not to define this method at all, but as we have properties that is not applied to Hahsable it required for impl.